StatsService.js

```diff
@@ -7,12 +7,119 @@ import axios from 'axios';
 
 const headers = {Accept: 'application/ld+json, application/json'};
 
-export class StatsService {
-  constructor() {
-  }
+/**
+ * A subscription is an async Function (such as ChartController.subscription)
+ * that is used to format the data for a Chart.
+ * @typedef {Promise} Subscription
+ * @param {Object} data - Data from the StatsService update.
+ * @param {Object} data.latest - The data from the latest update only.
+ * @param {Array<Object>} data.all - All of the data received so far.
+ */
+
+// instance stores the singleton
+let instance = null;
 
+class StatsService {
```

Comment:
We are going to change the singleton pattern here. The reason behind the change is because we want to allow for one to create instantiate new objects if they want. So, the change will require setting up a singleton in the `index.js` file and exporting that.

Here's an example of the pattern we want to achieve:
https://github.com/digitalbazaar/bedrock-web-store/blob/master/index.js

Reply:
that is beautiful and will do that first thing this morning.